pkg/nsx/services/inventory/build.go

```diff
@@ -90,7 +92,7 @@ func (s *InventoryService) BuildPod(pod *corev1.Pod) (retry bool) {
 
 	operation, _ := s.compareAndMergeUpdate(preContainerApplicationInstance, containerApplicationInstance)
 	if operation != operationNone {
-		s.pendingAdd[containerApplicationInstance.ExternalId] = containerApplicationInstance
+		s.pendingAdd[containerApplicationInstance.ExternalId] = &containerApplicationInstance
```

Comment:
Guess you also need to update the unit tests to use pointer for pendingAdd

Reply:
You're right. Done